### PR TITLE
Updated and pinned CF deps for CF-v1

### PR DESCRIPTION
**Pull request by jimmynotjim**
Updated the Capital Framework modules for version 1 so we can keep it up to date w/ the latest CF components.
## Removals
- Removed the `cf-concat` task because it's no longer needed
## Changes
- Added each CF module to main.less
- Updated less task to use relative paths
- Updated and pinned each CF module to the most current stable version (I'm not sure the Design Manual gets looked at enough by devs to ensure upstream changes don't break things)
## Testing
- Run the site locally, everything except the font bug outlined below should be the same
## Review
- design team (sorry, no global user)
- @Scotchester 
- @ascott1 
- @willbarton (cause I know he can run this site)
## Preview

screen shot 2015-07-30 at 12 28 12 pm

[Preview this PR without the whitespace changes](?w=0)
## Notes
- ~~There's a [bug in cf-core](https://github.com/cfpb/cf-core/pull/42) that's awaiting release for the medium and demi fonts~~
- This update will be necessary for the Design Manual to update to the latest [CF-Button changes](https://github.com/cfpb/design-manual/issues/263)

### CODE REVIEW

**Comment by willbarton** [comment]
cat

It works for me, as far as I can tell.

File: Gruntfile.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ module.exports = function(grunt) {
         options: {
           // The src/vendor paths are needed to find the CF components' files.
           // Feel free to add additional paths to the array passed to `concat`.
-          paths: grunt.file.expand('src/vendor/*').concat([])
+          paths: '<%= loc.src %>/vendor/'
```
Comment (Scotchester) [inline Gruntfile.js]: Can you be more specific about what you're doing here and why? Also, if this change remains, the comment above is out of date.

File: Gruntfile.js [hunk 2]
```diff
@@ -360,7 +360,7 @@ module.exports = function(grunt) {
   /**
    * Create custom task aliases and combinations.
    */
-  grunt.registerTask('compile-cf', ['bower:cf', 'concat:cf-less', 'concat:topdocIcons']);
+  grunt.registerTask('compile-cf', ['bower:cf', 'concat:topdocIcons']);
```
Comment (Scotchester) [inline Gruntfile.js:354]: If we're leaving the subtask in there, let's update the comment at `concat:cf-less` to explain that it's no longer/not currently used.

File: Gruntfile.js (hunk 1 above)
Comment (jimmynotjim) [inline Gruntfile.js]: Adding the vendor path means that you don't need the full path when importing with Less here https://github.com/cfpb/design-manual/pull/341/files#diff-32dca36d08c77c57a0a184ef881257eaR10.

Good catch on the comment, will remove

File: Gruntfile.js (hunk 2 above)
Comment (jimmynotjim) [inline Gruntfile.js:354]: Hmmm, maybe we should rename the `compile-cf` task because we're not actually compiling it any more.

Comment (Scotchester) [inline Gruntfile.js:354]: Probably, but that doesn't answer my original suggestion about `concat:cf-less` :)

File: Gruntfile.js (hunk 1 above)
Comment (Scotchester) [inline Gruntfile.js]: Isn't that what this was already doing, though? Is this just a simpler way of writing it? The expand+concat isn't needed after all?

Comment (jimmynotjim) [inline Gruntfile.js]: Essentially. Before it was creating a comma sep list of every individual cf-component dir. That's not necessary, you only need the root where they live (ie the vendor directory). Making it static like this is both clearer and less error prone.

File: Gruntfile.js (hunk 2 above)
Comment (jimmynotjim) [inline Gruntfile.js:354]: Oh now I see, I thought I had removed that concat task. Will update shortly.

Comment (jimmynotjim) [inline Gruntfile.js:354]: Removed the `concat:cf-less` task and updated the comment above to be JS specific.

**Comment by jimmynotjim** [comment]
@Scotchester could you take another look when you get a chance?

**Comment by Scotchester** [comment]
Yeah, I hope to have time to do it tomorrow.

**Comment by jimmynotjim** [comment]
@Scotchester if you get a chance this week, can you try the updated solution to the import issue? If everything is back to normal I'll squash this so it can be merged.

**Comment by Scotchester** [comment]
Gave this another try today. After pulling the latest of your branch and running `grunt`, everything in `/static/js/` still changed :\

**Comment by jimmynotjim** [comment]
Switch to Gulp? :trollface: I'm seriously running out of ideas.

**Comment by KimberlyMunoz** [comment]
I compared the PR with what happens when I run `grunt` and the difference is that new JS for cf-expandables is being added when I run grunt, but those lines are not being included in the generated files in this PR.

**Comment by jimmynotjim** [comment]
Oh oh oh, @Scotchester you had changes in `/static/js/`? The issue we were having before was the css changing. Thanks for the clue @KimberlyMunoz, I'll look into what's causing this.

**Comment by jimmynotjim** [comment]
@Scotchester @KimberlyMunoz try again if you could? I think I've got everything tidied up

**Comment by KimberlyMunoz** [comment]
Now the CSS is changing! I'll take a look and report back.

**Comment by KimberlyMunoz** [comment]
It looks like it's something with the autoprefixing. The left is what's in your PR, the right is what I generate.

image
image

**Comment by cfarm** [comment]
Mine works, please review: #359

**Comment by Scotchester** [comment]
Closing in favor of #359 for now. Once the DM surge is complete, I'd like to revisit the notion of pinning the individual deps, as updates will be much fewer and farther between.

**Pull request closed without merging**